src/libs/EmojiUtils.js

```diff
@@ -159,30 +168,38 @@ function mergeEmojisWithFrequentlyUsedEmojis(emojis, frequentlyUsedEmojis = [])
 
 /**
  * Update the frequently used emojis list by usage and sync with API
- * @param {Object[]} frequentlyUsedEmojis
- * @param {Object} newEmoji
+ * @param {Object|Object[]} newEmoji
  */
-function addToFrequentlyUsedEmojis(frequentlyUsedEmojis, newEmoji) {
-    let frequentEmojiList = frequentlyUsedEmojis;
-    let currentEmojiCount = 1;
+function addToFrequentlyUsedEmojis(newEmoji) {
     const currentTimestamp = moment().unix();
-    const emojiIndex = _.findIndex(frequentEmojiList, e => e.code === newEmoji.code);
-    if (emojiIndex >= 0) {
-        currentEmojiCount = frequentEmojiList[emojiIndex].count + 1;
-        frequentEmojiList.splice(emojiIndex, 1);
-    }
-    const updatedEmoji = {...newEmoji, ...{count: currentEmojiCount, lastUpdatedAt: currentTimestamp}};
-    const maxFrequentEmojiCount = (CONST.EMOJI_FREQUENT_ROW_COUNT * CONST.EMOJI_NUM_PER_ROW) - 1;
+    const maxFrequentEmojiCount = (CONST.EMOJI_FREQUENT_ROW_COUNT * CONST.EMOJI_NUM_PER_ROW);
+
+    // Get unique emojis array with counts for every emoji, these emojis are usually extracted from copy/pasted comments
+    const uniqueEmojisWithCounts = _.chain([].concat(newEmoji))
+        .groupBy('name')
+        .map(values => ({
+            ...values[0],
+            count: values.length,
+            lastUpdatedAt: currentTimestamp,
+        }))
+        .value();
+
+    // Concat uniqueEmojisWithCounts with the frequentlyUsedEmojis where we sum the count and update the lastUpdatedAt
+    const mergedEmojisWithFrequent = _.chain(uniqueEmojisWithCounts)
+        .concat(frequentlyUsedEmojis)
+        .groupBy('name')
+        .map(groupedEmojiList => ({
+            ...groupedEmojiList[0],
+            count: lodashSumBy(groupedEmojiList, 'count'),
+            lastUpdatedAt: lodashMaxBy(groupedEmojiList, 'lastUpdatedAt').lastUpdatedAt,
+        }))
+        .value();
 
-    // We want to make sure the current emoji is added to the list
-    // Hence, we take one less than the current high frequent used emojis and if same then sorted by lastUpdatedAt
-    frequentEmojiList = lodashOrderBy(frequentEmojiList, ['count', 'lastUpdatedAt'], ['desc', 'desc']);
-    frequentEmojiList = frequentEmojiList.slice(0, maxFrequentEmojiCount);
-    frequentEmojiList.push(updatedEmoji);
+    // Sort the list and take the first maxFrequentEmojiCount items
+    const frequentEmojiListOrdered = lodashOrderBy(mergedEmojisWithFrequent, ['count', 'lastUpdatedAt'], ['desc', 'desc'])
+        .slice(0, maxFrequentEmojiCount);
```

Comment:
It works fine for me though. Can you share your repro steps?
https://user-images.githubusercontent.com/50919443/236589877-7beb7539-d630-475d-8023-9691afcc8452.mov

Reply:
I'm not aware the video I attached is a link. updated.

The emoji I add is a new emoji (the monkey emoji), not an emoji with count > 1

Reply:
Ah, I think I understand.

Reply:
I don't think the fix above is enough because it could also happen when we insert multiple new emojis, right? I think the old logic that splice every found emoji is the best way for this.

Reply:
Or I was thinking we can use something like _.without, but it doesn't accept a parameter for the comparator. I was hoping we can do something like _chain().without(uniqueEmojisWithCounts, 'name') to remove inserted emoji from the list, then later on we push it back.

Reply:
Actually, I think our current logic is the correct one, no? (Slack did the same thing) If the frequent emoji list count is all > 1 and we insert a new emoji, would we want to replace the emoji with count > 1? Imagine if the user use some emojis a lot (> 100), then when they just try to use a new emoji, it replaces the emojis with bigger count.

Reply:
Tested the new logic and works well.

> Actually, I think our current logic is the correct one, no? (Slack did the same thing)

Ask about this one?

Reply:
After keep thinking about it, I think we should keep the old behavior because we only keep track the count of the emoji inside the frequent list, which means no matter how many times we insert the new emoji, it won't added to the list (because the count is always 0).

Reply:
My bad. That's what will happen with the new behavior. Updated my comment.

Reply:
The count on the frequent emoji list. Let say all 24 emoji has count of 2. Then, we add a new emoji. Because it doesn't exist in the list, it will have a count of 1. With our new logic, we won't take the new emoji. The next time we add the same emoji, the count will still be 1 because it doesn't exist in the list.

Reply:
Updated to solve the issue. Also added a test.